### PR TITLE
Update IOutputAllocator usage to be compatible with TensorRT 9.2 APIs

### PR DESCRIPTION
**Pull request by jhalakpatel**
`IOutputAllocator` needs to be supported for both TRT 9.2 and TRT 10.x release. Add support for using TRT 9.2 API.

### CODE REVIEW

File: mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp [hunk 1]
```diff
@@ -155,6 +164,30 @@ class OutputAllocatorImpl : public nvinfer1::IOutputAllocator {
     }
     return reinterpret_cast<void *>(mOutputPtr);
   }
+#else
+  void *reallocateOutput(const char *name, void *memory, uint64_t size,
```
Comment (shelkesagar29) [inline mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp:168]: Does this work with TRT 8.6 too?

File: mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp [hunk 2]
```diff
@@ -37,6 +37,14 @@
 #pragma GCC diagnostic pop
 #endif
 
+/// Evaluates to true if the version of TensorRT that we are compiling against
+/// is greater than or equal to the given version number.
+#define TENSORRT_MODULE_VERSION_GTE(major, minor, patch)                       \
```
Comment (shelkesagar29) [inline mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp:42]: Can we use `MLIR_TRT_COMPILE_TIME_TENSORRT_VERSION_GTE` that is already available?

Comment (jhalakpatel) [reply mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp:42]: I was running into some compilation issues while including `mlir-tensorrt/tensorrt/include/mlir-tensorrt-dialect/Utils/TensorRTVersion.h` here.
I did not investigate the compilation issue further, but does that dependency make sense in the `mlir-tensorrt/executor`? If yes, I can update the Makefile to add the dependency.

File: mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp (hunk 1 above)
Comment (jhalakpatel) [reply mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp:168]: Yes, should work with TRT 8.6 as well.

File: mlir-tensorrt/executor/test/IntegrationTests/lua_results.mlir [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// RUN: rm %t.mlir || true
```
Comment (jhalakpatel) [inline mlir-tensorrt/executor/test/IntegrationTests/lua_results.mlir:1]: Please ignore this file. I will remove it soon.

File: mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp (hunk 2 above)
Comment (shelkesagar29) [reply mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp:42]: This should depend only on `MLIRTRTTensorRTDynamicLoader`, I think. Adding dependency is okay.

Comment (jhalakpatel) [reply mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp]: I think I needed to explicitly include a header directory. Link time dependency is not required.

**Review by shelkesagar29 — APPROVED** [review]

**Pull request merged**